### PR TITLE
Support for `flight support`

### PR DESCRIPTION
**Pull request by benarmston**
This PR adds support for the first phase of the `flight support` CLI tool.

The `flight support` tool is a cluster CLI tool designed to help site viewers and site contacts find help using their cluster.  In its initial phase, `flight support` provides a knowledge base of topics and articles that the user can browse through.  Later phases will improve this by adding search facilities.  Eventually `flight support` will allow users to log support requests in a similar manner to the flight center's case form.

To support the initial phase, Flight Center provides two new models: `Topic` and `Article`.  An `Article` consists of a `title`, some `metadata` and an `url`.  The `url` points to an, err..., article on some aspect of using an Alces Flight cluster, initially based on those available at http://docs.alces-flight.com.  `Article`s are grouped under `Topic`s to make browsing the list of `Article`s more pleasant.  `Topic`s contain a `title`, a list of `Article`s and optionally belong to a `Site`.

If a `Topic` does not belong to a site, it is considered to be a global `Topic` and all of its `Article`s are likewise global.  Global `Topic`s and `Article`s are available to all site contacts, site viewers and admins.  If a `Topic` does belong to a site, it and all of its articles are only available to that site's users.

There is currently no support for overriding global articles with site-specific ones. Or adding site-specific articles to a global topic.

### CODE REVIEW

**Comment by benarmston** [comment]
@jamesremuscat issues with travis and clahub are now resolved.

File: config/topics.yml [hunk 1]
```diff
@@ -0,0 +1,172 @@
+- title: Using your cluster
```
[the remark this answers is not in the document]
Comment (benarmston) [reply config/topics.yml]: Good idea.

File: db/migrate/20180716143624_create_articles.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class CreateArticles < ActiveRecord::Migration[5.2]
+  def change
+    create_table :articles do |t|
+      t.string :title, null: false, limit: 255
+      t.string :url, null: false, limit: 512
+      t.json :meta, null: false, default: '{}'
+      t.references :topic, foreign_key: true
```
[the remark this answers is not in the document]
Comment (benarmston) [reply db/migrate/20180716143624_create_articles.rb]: Good catch.

[the remark this answers is not in the document]
Comment (benarmston) [reply db/migrate/20180716143624_create_articles.rb]: FYI: migration updated to include `null: false`.

File: config/topics.yml [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+  articles:
+    - title: Basic cluster operation
+      url: https://raw.githubusercontent.com/alces-software/flight-appliance-docs/2017.2r1/docs/source/basics/basic_cluster_operation.rst
+      meta:
+        section: 5
```
[the remark this answers is not in the document]
Comment (benarmston) [reply config/topics.yml]: Man pages belong in different sections.  This is used when searching for a man page, either with `man foo` or via `man -k bar`.  The section is also displayed on the first line of the rendered man page, e.g., `LS(1)` or `PASSWD(5)`.  If a man page does not specify a section, the section is omitted but not the parentheses, e.g., `Bad man page()`.

The available sections are:

```
       1   Executable programs or shell commands
       2   System calls (functions provided by the kernel)
       3   Library calls (functions within program libraries)
       4   Special files (usually found in /dev)
       5   File formats and conventions eg /etc/passwd
       6   Games
       7   Miscellaneous  (including  macro  packages  and  conventions), e.g.
           man(7), groff(7)
       8   System administration commands (usually only for root)
       9   Kernel routines [Non standard]
```

I think most (all?) or the pages I've added best fall under section 5 conventions.  I'm willing to listen to suggestions for alternative section numbers.

[the remark this answers is not in the document]
Comment (mjtko) [reply config/topics.yml]: Think we used `7` previously.

(Tho I quite like the idea of using `6`. 😉)

**Comment by benarmston** [comment]
@jamesremuscat I've addressed / commented on all points in your review.

File: config/topics.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jamesremuscat) [reply config/topics.yml]: Ah, I hadn't twigged that it was man pages. That makes sense :slightly_smiling_face:

[the remark this answers is not in the document]
Comment (benarmston) [reply config/topics.yml]: Perhaps we should compromise with `6.5`.  Man seems quite happy with that.  (I'll change them to `7` that seems a better fit than `5`).

**Comment by jamesremuscat** [comment]
Rebased on latest `develop` - quite a few migrations had been added since the ones in this branch were made so the process was a little more involved than it sometimes is!

**Pull request merged**